src/db.rs

```diff
@@ -179,6 +182,51 @@ pub async fn run_migrations(client: &DbClient) -> anyhow::Result<()> {
     Ok(())
 }
 
+pub async fn run_scheduled_jobs(db: &DbClient) -> anyhow::Result<()> {
+    let jobs = get_jobs_to_execute(&db).await.unwrap();
+    println!("jobs to execute: {:#?}", jobs);
+    tracing::trace!("jobs to execute: {:#?}", jobs);
+
+    for job in jobs.iter() {
+        update_job_executed_at(&db, &job.id).await?;
+
+        match handle_job(&job.name, &job.metadata).await {
+            Ok(_) => {
+                println!("job succesfully executed (id={})", job.id);
+                tracing::trace!("job succesfully executed (id={})", job.id);
+
+                if let Some(frequency) = job.frequency {
```

Comment:
I think the front reinsert should probably be driven regardless of failure, and we'll just want to be careful about having too frequent cron tasks. Otherwise we turn transient failure into a permanent one.

Reply:
Yeah makes sense, will move that outside the successful case. Thanks !